kubectl-plugin/pkg/cmd/create/create_cluster.go

```diff
@@ -0,0 +1,223 @@
+package create
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/ray-project/kuberay/kubectl-plugin/pkg/util/generation"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type CreateClusterOptions struct {
+	configFlags    *genericclioptions.ConfigFlags
+	ioStreams      *genericclioptions.IOStreams
+	clusterName    string
+	fileName       string
+	rayVersion     string
+	image          string
+	headCPU        string
+	headMemory     string
+	workerGrpName  string
+	workerCPU      string
+	workerMemory   string
+	workerReplicas int32
+}
+
+var (
+	createClusterLong = templates.LongDesc(`
+		Creates Ray Cluster from inputed file or generate one for user.
+	`)
+
+	createClusterExample = templates.Examples(`
+		# Creates Ray Cluster from inputted Yaml
+		kubectl ray create cluster -f raycluster-stample.yaml
+
+		# Creates Ray Cluster from flags input
+		kubectl ray create cluster sample-cluster --ray-version 2.9.0 --image rayproject/ray:2.9.0 --head-cpu 1 head-memory 5Gi --worker-grp-name worker-group1 --worker-replicas 3 --worker-cpu 1 --worker-memory 5Gi
```

Comment:
Should we add a `--dry-run` or `-o yaml` option to print the generated YAML from this command?

Reply:
Ah, I see from the implementation that we're already printing the generated YAML

Reply:
would having `--dry-run` be a better option than just printing out the yaml and waiting for user input?

Reply:
> would having --dry-run be a better option than just printing out the yaml and waiting for user input?

Yeah maybe -- by default the output should just be "Created Ray cluster ...." and `--dry-run` can output the YAML? Not sure we need to print the YAML every time

Reply:
I've added a --dry-run flag that will print out the yaml if its set and will just apply the yaml if unset.